### PR TITLE
QR-3016 better log formatting and fix error location detection

### PR DESCRIPTION
**Pull request by git-jiby-me**
=> move stack traces from errors to top level stack_trace attribute
=> indicate if an exception was uncaught one
=> calculate the error location from trace in case of uncaught
exceptions

### CODE REVIEW

File: lib/formatter.js [hunk 1]
```diff
@@ -13,36 +13,34 @@ var CALLER_UPPER_TRACE_FUNCTION = 'Module._compile';
  * @return {string}         serialized json log
  */
 function formatLog (qlogger, options) {
+    // if log was already formatted, directly return already formatted log
```
Comment (leonkyr) [inline lib/formatter.js]: Please add unit tests for this module

Comment (git-jiby-me) [inline lib/formatter.js]: Ok, will do :+1:

**Comment by git-jiby-me** [comment]
@leonkyr unit tests added.

File: lib/formatter.js [hunk 2]
```diff
@@ -54,6 +52,52 @@ function formatLog (qlogger, options) {
 }
 
 /**
+ * get location where error occured from stack trace
+ * @param  {string} stack stack trace
+ * @return {Object}       error location information
+ */
+function getErrorLocationInfoFromErrorStack (stack) {
+    var callerInfo = stack.split('at ')[1].trim().split(' ');
```
Comment (leonkyr) [inline lib/formatter.js]: This line required some more explanation in the code.. what is [1]?

File: lib/formatter.js [hunk 3]
```diff
@@ -54,6 +52,52 @@ function formatLog (qlogger, options) {
...
+    var fileInfo = callerInfo[callerInfo.length - 1];
+    // jscs:disable requireCamelCaseOrUpperCaseIdentifiers
+    // jshint camelcase: false
+    return {
+        function_name: callerInfo.length === 2 ? callerInfo[0] : null,
+        file_name: fileInfo.split(':')[0],
```
Comment (leonkyr) [inline lib/formatter.js]: why do we need to do this splits?

Comment (git-jiby-me) [inline lib/formatter.js]: in the stack trace we have information as string like
"at simulateError qlogger/lib/formatter.spec.js 41:23" in the format of
"at function_name file_name line_number:column:number".

we split to extract these parameters

File: lib/formatter.js (hunk 2 above)
Comment (git-jiby-me) [inline lib/formatter.js]: ok, i will add an inline comment to make it clear

**Pull request merged**